iohub/ngff/nodes.py

```diff
@@ -1930,14 +1937,56 @@ def rename_well(self, old: str, new: str):
         self.dump_meta()
 
 
+def _check_file_mode(
+    store_path: Path,
+    mode: Literal["r", "r+", "a", "w", "w-"],
+    disable_path_checking: bool,
+) -> bool:
+    if mode == "a":
+        mode = ("w-", "r+")[int(store_path.exists())]
+    parse_meta = False
+    if mode in ("r", "r+"):
+        parse_meta = True
+    elif mode == "w-":
+        if store_path.exists():
+            raise FileExistsError(store_path)
+    elif mode == "w":
+        if store_path.exists():
+            if (
+                ".zarr" not in str(store_path.resolve())
```

Comment:
Does `store_path` contain an output key as well? e.g., would `/foo/bar/baz.zarr/qux` count? Or do we want the path to end with `.zarr`?

Reply:
I think things like `plate.zarr/row/column/fov` should count.